calculator.js

```diff
@@ -1,126 +1,134 @@
-function Calculator() {
-    'use strict';
-
-    let newInput = "0";
-    let oldInput = "";
-    let result = "";
-    this.input = ko.observable(newInput);
-    let operation = "";
-    let history = [];
-
-    //--------ADD NUMBER---------
-    this.addNumber = (number) => () => {
-        if (newInput === "0") {
-            newInput = number.toString();
+(function (factory) {

```

Comment:
In order to test our application code, we had 2 choices. One of those was to run the tests in the browser. Another option was to run the tests in Node.JS. I think it's better that we run our tests in Node.JS. However, there are several problems associated with that. The browser and Node JS are very different execution environments. The browser would typically have knockout loaded as a global variable "ko". But Node JS would need it to be required in as a module. Also the browser would simply assign Calculator to a global variable. But Node JS would need it to be a module. We can't do `require('../calculator')` in our unit test file because nothing gets exported.

For this reason, we put our Calculator in a function called a "factory function". Our "factory function" takes as parameters the dependencies it has (in our case, knockout) and it returns our module (in our case, Calculator). Then we pass our factory function into an IIFE which will run the function, passing it knockout, and doing whatever we want with the result of the function (our Calculator). I will illustrate more in the rest of the comments.